src/consensus/aft/orc_requests.h

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#pragma once
+
+#include "ds/thread_messaging.h"
+#include "kv/kv_types.h"
+#include "node/node_client.h"
+#include "node/rpc/serdes.h"
+
+namespace ccf
+{
+  struct ObservedReconfigurationCommit
+  {
+    struct In
+    {
+      ccf::NodeId from;
+      kv::ReconfigurationId reconfiguration_id;
+    };
+
+    using Out = void;
+  };
+
+  DECLARE_JSON_TYPE(ObservedReconfigurationCommit::In)
+  DECLARE_JSON_REQUIRED_FIELDS(
+    ObservedReconfigurationCommit::In, from, reconfiguration_id)
+
+  inline bool submit_orc(
+    std::shared_ptr<ccf::NodeClient> client,
+    const ccf::NodeId& from,
+    kv::ReconfigurationId rid)
+  {
+    LOG_DEBUG_FMT("Configurations: submit ORC for #{} from {}", rid, from);
+
+    ObservedReconfigurationCommit::In ps = {from, rid};
+
+    http::Request request(fmt::format(
+      "/{}/{}", ccf::get_actor_prefix(ccf::ActorsType::nodes), "orc"));
+    request.set_header(
+      http::headers::CONTENT_TYPE, http::headervalues::contenttype::JSON);
+
+    auto body = serdes::pack(ps, serdes::Pack::Text);
+    request.set_body(&body);
+    return client->make_request(request);
+  }
+
+  struct AsyncORCTaskMsg
+  {
+    AsyncORCTaskMsg(
+      std::shared_ptr<ccf::NodeClient> client_,
+      const ccf::NodeId& from_,
+      kv::ReconfigurationId rid_,
+      size_t retries_ = SIZE_MAX) :
+      client(client_),
+      from(from_),
+      rid(rid_),
+      retries(retries_)
+    {}
+
+    std::shared_ptr<ccf::NodeClient> client;
+    ccf::NodeId from;
+    kv::ReconfigurationId rid;
+    size_t retries;
+  };
+
+  inline void orc_cb(std::unique_ptr<threading::Tmsg<AsyncORCTaskMsg>> msg)
+  {
+    if (!submit_orc(msg->data.client, msg->data.from, msg->data.rid))
+    {
+      if (--msg->data.retries > 0)
```

Comment:
Why is there a limit here? If the node never sends the ORC, will the node be stuck forever?

Reply:
Yes, it would be stuck. The limit is `SIZE_MAX` though. In the beginning I wasn't sure whether we need/want a limit, but you're right, we should keep trying forever.